server/api/socket/asset_manager.py

```diff
@@ -115,6 +115,10 @@ async def assetmgmt_rm(sid, data):
     if asset.owner != user:
         logger.warning(f"{user.name} attempted to remove a file it doesn't own.")
         return
+    if asset.file_hash is not None and (ASSETS_DIR / asset.file_hash).exists():
+        unique_asset_owners = Asset.select(Asset.owner).where(Asset.file_hash == asset.file_hash).distinct()
```

Comment:
You can use `.count()` I believe at the end to do the count in database, so no need to do the length check below.

I can see why you check distinct on the asset owner, but I think you just want to make sure there isn't any other asset that links to it irregardless of the owner.  Suppose you've uploaded a map to two folders and decide you don't need one of them, deleting the one in folder x/y/z should not affect the one in a/b/c.

Reply:
Yeah, that makes sense, I was trying to be too specific and it cluttered the code and didn't really work 😄 I'll take another shot at it